programs/voter-stake-registry/src/instructions/create_deposit_entry.rs

```diff
@@ -17,9 +18,24 @@ pub struct CreateDepositEntry<'info> {
         has_one = registrar,
         has_one = voter_authority)]
     pub voter: AccountLoader<'info, Voter>,
+
+    #[account(
+        init_if_needed,
+        associated_token::authority = voter,
+        associated_token::mint = deposit_mint,
+        payer = voter_authority
+    )]
+    pub vault: Box<Account<'info, TokenAccount>>,
+
+    #[account(mut)]
     pub voter_authority: Signer<'info>,
```

Comment:
I'm unsure if it's good form to separate the authority and the payer. Probably ok not to?

Reply:
we do this separation in `create_voter`, so I'd also do it here